src/renderer/commands/createDrawObject.ts

```diff
@@ -8,36 +9,36 @@ import REGL = require('regl');
 
 // Uniforms are the same for all vertices.
 interface Uniforms {
-    projection: REGL.Mat4;
-    view: REGL.Mat4;
-    model: REGL.Mat4;
+    projection: mat4;
+    view: mat4;
+    model: mat4;
     numLights: number;
-    ambientLight: REGL.Vec3;
-    lightPositions: REGL.Vec3[];
```

Comment:
What's the deal with these properties? Were they just duplicated before?

Reply:
For uniform arrays like `lightPositions`, `lightColors`, and `lightIntensities`, we don't actually have a single key called `lightPositions`. We have `lightPositions[0]`, `lightPositions[1]`, etc. We don't know how many at compile time, so we can't specify these in the interface.

Reply:
So were we not using these before? I understand why we can't put them here anymore, but I'm not sure I understand why they were here before. Was that also something we missed because of no regl type-checking?

Reply:
We do use them, basically this interface defines everything that we are required to pass into the `uniforms: { }` object farther down when we are creating the draw command. We always specified values for the light position uniform, but we never had a key in the object named exactly `lightPositions`, only `lightPositions[n]`, so when we turned on the type checking, it was complaining that we need to pass in `lightPositions`. So now, by removing that from the `Uniforms` interface, we're just not typechecking `lightPositions` at all because there's no way to statically check this when we have dynamic keys.

Reply:
Ah, got it.